### ISSUE TITLE
dplyr::filter giving inconsistent results when source-ing a script

### ISSUE THREAD

**Issue by stephenjjohnson** [hadley/dplyr#1899]
If I run the following code line by line it gives the expected output of df1 - a data frame containing 2 observations of 4 variables, and df2 - 3 obs of 2 vars.

However, if I source the entire script then the `select` operation works as expected but the `filter` operation gives a df1 containing 0 observations.

I have confirmed this behaviour in RStudio 0.99.902 with R 3.3.0 and dplyr 0.4.3 on Win7 (32bit), Win10 (64bit) and Ubuntu 16.04 LTS (64-bit).

I also tried prepending the operations with `dplyr::` but it did not change the behaviour.

    library(dplyr)
    df <- read.csv(text = '"a", "b", "c", "d"
                   "beef", 1, 2, 3
                   "beef", 4, 5, 6
                   "chicken", 7, 8, 9')
    df1 <- df %>% filter(a == "beef")
    df2 <- df %>% select(c(a, b))

**Comment by hadley** [comment]
How are you sourcing the script?

**Comment by ShuguangSun** [comment]
df$a is a factor, and levels(df$a) show

> [1] "               beef"    "               chicken"

There are spaces.

**Comment by stephenjjohnson** [comment]
@hadley I am using the Source button at the top of the editor pane in RStudio.

@ShuguangSun I checked `levels(df$a)` when stepping through the lines with the Run button:

> [1] "beef"    "chicken"

and again after Source-ing the whole script

> [1] "    beef"    "    chicken"

with spaces - as you noted.

So - I deleted the tabs/spaces from the script:

    library(dplyr)
    df <- read.csv(text = '"a", "b", "c", "d"
    "beef", 1, 2, 3
    "beef", 4, 5, 6
    "chicken", 7, 8, 9')
    df1 <- df %>% dplyr::filter(a == "beef")
    df2 <- df %>% dplyr::select(c(a, b))

and Source it again. Now I get the right result.

So - it looks like R is inserting spaces when sourcing the whole script with whitespace, and it is not a problem with `dplyr`.
